[14.0] [MIG] base_jsonify module

chafique-delli opened the pull request.

sebastienbeau committed:
add base_jsonify module

lmignon committed:
base_jsonify: Allow fields aliasing in jsonify

lmignon committed:
base_jsonify: typo

sebastienbeau committed:
base_jsonify: add support of reference fields

sebastienbeau committed:
base_jsonify: remove False for int, float, string

hparfr committed:
base_jsonify: cleanups

remove copyrights from init and update copyrights

lmignon committed:
base_jsonify: Output 'false' into json only for boolean fields

sebastienbeau committed:
base_jsonify: restore travis configuration (#112)

* [ADD] restore travis configuration

* [PEP] fix pep8

* [REF] exclude  product_categ_available_pos of testing due to issue https://github.com/odoo/odoo/pull/23749

Commit:
base_jsonify: Use readme fragment mechanism

legalsylvain committed:
base_jsonify: add view for ir.exports

simahawk committed:
[12.0][MIG] base_jsonify: py3

Commit:
base_jsonify: misc fixes on comments/docstrings

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update base_jsonify.pot

simahawk committed:
[12.0][MIG] base_jsonify: cleanup

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update base_jsonify.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (11 of 11 strings)

Translation: server-tools-12.0/server-tools-12.0-base_jsonify
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-base_jsonify/zh_CN/

lmignon committed:
[FIX]base_jsonify: Preserve initial order on ir.export.line
We can't change the order on ir.export.line since it's used to preserve the order speicifed by the user when creating a export filter into the UI

oca-travis committed:
[UPD] Update base_jsonify.pot

OCA-git-bot committed:
base_jsonify 12.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-base_jsonify
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-base_jsonify/

Commit:
[IMP] base_jsonify: black, isort

Commit:
[MIG] base_jsonify: Migration to 13.0

oca-travis committed:
[UPD] Update base_jsonify.pot

OCA-git-bot committed:
[UPD] README.rst

lmignon committed:
[FIX] base_jsonify: In odoo 12, Date and Datetime fields values are datetime object
Convert fields.Date and fields.Datetime values to string when serializing to json

OCA-git-bot committed:
base_jsonify 13.0.1.0.1

lmignon committed:
[IMP] base_jsonify: Serialize Date and Datetime into ISO 8601 format

lmignon committed:
[FIX] base_jsonify: Add tz info into serialized datetime

lmignon committed:
[FIX] base_jsonify: Serialize datetime into the user's timezone

lmignon committed:
base_jsonify 13.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-13.0/server-tools-13.0-base_jsonify
Translate-URL: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-base_jsonify/

OCA-git-bot committed:
pre-commit update

simahawk committed:
base_jsonify: ease override jsonify per record

simahawk committed:
base_jsonify: support callable parser

simahawk committed:
base_jsonify: tests use SavepointCase

oca-travis committed:
[UPD] Update base_jsonify.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
base_jsonify 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-13.0/server-tools-13.0-base_jsonify
Translate-URL: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-base_jsonify/

Commit:
[IMP] base_jsonify: add lang, custom resolvers, alias* support

Commit:
[IMP] base_jsonify: simplify full parser API when not using translations

Commit:
[IMP] base_jsonify: add context to resolver eval method

simahawk committed:
base_jsonify: allow export line exclusion

Sometimes you just want to turn off via configuration
lines created by some other modules.

simahawk committed:
base_jsonify: return one record on demand

If you use this module extensively - as I do -
you've done this tons of time. For sure :)
This little change simplifies your code base where needed.

oca-travis committed:
[UPD] Update base_jsonify.pot

OCA-git-bot committed:
base_jsonify 13.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-13.0/server-tools-13.0-base_jsonify
Translate-URL: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-base_jsonify/

OCA-git-bot committed:
base_jsonify 13.0.1.3.0

Commit:
[IMP] base_jsonify: better help message in resolvers

Commit:
[REF] base_jsonify: rename ir.exports.line field `alias` to `target`

Commit:
[TEST] base_jsonify: test and document code

chafique-delli committed:
[MIG] base_jsonify: Migration to 14.0

lmignon committed:
[IMP] base_jsonify: Allow to export many2one and referene field as simple field

Before this change, the result of the export of a relational field was always a dict or a list of dict.

ex:

<record id="ir_exp_shopinvader_variant_lang" model="ir.exports.line">
  <field name="name">publication_language_id/name</field>
  <field name="alias">publication_language_id/name:lang</field>
  <field name="export_id" ref="shopinvader.ir_exp_shopinvader_variant" />
</record>

will output:
{
    ...
    "publication_language_id" : {
        "lang": "French"
    }
    ...
}

After this change it's now possible to define simple exporter for relational fields where the value into the result will be the display_name or a list of display_name of the related records.

ex:

<record id="ir_exp_shopinvader_variant_lang" model="ir.exports.line">
  <field name="name">publication_language_id</field>
  <field name="alias">publication_language_id:lang</field>
  <field name="export_id" ref="shopinvader.ir_exp_shopinvader_variant" />
</record>

will output:
{
    "lang": "French"
}
 Please enter the commit message for your changes. Lines starting

simahawk commented on base_jsonify/readme/CONTRIBUTORS.rst:
```diff
@@ -1,3 +1,4 @@
 * BEAU Sébastien <sebastien.beau@akretion.com>
 * Raphaël Reverdy <raphael.reverdy@akretion.com>
 * Laurent Mignon <laurent.mignon@acsone.eu>
+* Chafique Delli <chafique.delli@akretion.com>
```
plain one 2 one migration, not sure this is needed :)

simahawk reviewed (commented):
LG, would you mind pls giving a review to https://github.com/OCA/server-tools/pull/1894 which would be nice to include into v14 as well?

OCA-git-bot committed:
base_jsonify 13.0.1.3.1

hparfr commented:
Hi Chafique, can you please add this one too https://github.com/OCA/server-tools/pull/1912 ?

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-13.0/server-tools-13.0-base_jsonify
Translate-URL: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-base_jsonify/

sebastienbeau commented:
ping @chafique-delli

lmignon commented:
Hi @chafique-delli 

Thank you for the migration... As @simahawk I think it would be nice to give a review to #1894  to include it into V14. 
Otherwise LGTM.

Commit:
[REF] base_jsonify: review changes

Commit:
[IMP] base_jsonify: cache json parser

oca-travis committed:
[UPD] Update base_jsonify.pot

OCA-git-bot committed:
base_jsonify 13.0.3.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-13.0/server-tools-13.0-base_jsonify
Translate-URL: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-base_jsonify/

simahawk commented:
/ocabot merge nobump
